skbio/diversity/_base.py

```diff
@@ -87,3 +88,18 @@ def _validate_otu_ids_and_tree(counts, otu_ids, tree):
         raise MissingNodeError("All otu_ids must be present as tip names in "
                                "tree. Tree is missing tips with names: %s"
                                % " ".join(missing_tip_names))
+
+
+def _vectorize_counts_and_tree(counts, otu_ids, tree, tree_index=None):
```

Comment:
Let's kill ``tree_index`` as a parameter for now, since it's not being used currently (it's weird to have have a path through this private function that we never hit).